### ISSUE TITLE
CRAN resubmission

### ISSUE THREAD

**Issue by martinju** [NorskRegnesentral/shapr#230]
Comments by CRAN after 2nd submission

- [X] 1. Please add small executable examples in your Rd-files to illustrate the
use of the exported function. [Took the occasion to clean up what are exported/non-exported functions and internal/non-internal functions. Now just a few are non-internal and they all have examples. Done over a range of commits]

- [x] 2. Please ensure that your functions do not write by default or in your
examples/vignettes/tests in the user's home filespace (including the
package directory and getwd()). This is not allowed by CRAN policies.
In your examples/vignettes/tests you can write to tempdir().
e.g. vignette [Asking CRAN for more information]

- [x] 3. Please do not modifiy the .GlobalEnv. This is not allowed by the CRAN
policies. e.g. rm(list = ls()) in vignette [I can't find anything in the vignette that changes .GlobalEnv, or in any package functions. I did find a script that uses rm(list=ls()), so just deleted that. Hopefully that's it dca25c6]

------

Have not checked what code I causing these issues, but will look at it tomorrow and ask for help if I can't figure out what the reviewer refers to.

**Comment by martinju** [comment]
Regarding 2. I can't find any hits on save, write or similar in functions, vignettes or tests. Could it be the use of expect_known_value? This writes to disk (updates the file) if the file is missing or changes. If this is it, we may get out of it by disabling automatic updating.

Note: There are indeed scripts under inst/scripts that writes to disk, but there is no mentioning of that not being allowed...

Any idea @nikolase90 ?

**Comment by nikolase90** [comment]
I'll take a look 👍

**Comment by martinju** [comment]
This got a little messy commit-wise, so here are all changes I did for this resubmission:

https://github.com/NorskRegnesentral/shapr/pull/220/files/bbbe005c47b011c1be68710a298e3d76066175df..e0bdd5e783b67662f01e1292bc5482cfa638743b

This is the update to cran-comments, summarizing what I did

* Revised what is specified as internal/non-internal functions, and added 
  examples such that all non-internal exported functions now have examples.
* I could not find any writing to disk in examples/vignettes/tests as suggested.
  There are scripts under inst/scripts which writes to disk. These scripts are 
  creating objects which are loaded in the tests and therefore handy for 
  package development. Note that we are also using testthat::expect_known_value
  in our tests, this function is overwriting .rds-objects if and only if
  the test fails. Please let me know more precisely what the issue is, if it
  persists after this re-submission.
* Deleted a call to rm(list=ls()) in a script under inst/scripts. I could not
  find any other modifications of .GlobalEnv in the vignette as suggested.

Additonal NOTE: 
* checking for future file timestamps ... NOTE
  unable to verify current time
> This is a known issue with the worldclockapi.com currently being down
  (https://stackoverflow.com/questions/63613301/r-cmd-check-note-unable-to-verify-current-time)

_____

So basically, I think I did what was suggested for point 1 and 3, also fixed the comment by @nikolase90 about unncessary use shapr:: for exported function on two occasions. 
For point 2 I don't know what is wrong, so I suggest to just resubmit, asking them to be more specific if they still think point 2 is an issue. 

Re-running all tests now.

**Comment by martinju** [comment]
> I'll take a look 👍

Great! See my last message above. I am ready to resubmit unless you or the final checks suggests we should change something.

**Comment by nikolase90** [comment]
What about just adding `inst/scripts/` to `.Rbuildignore`? Is there any reason that the scripts should be part of the package?

**Comment by martinju** [comment]
> What about just adding `inst/scripts/` to `.Rbuildignore`? Is there any reason that the scripts should be part of the package?

Certainly possible. I think it is neat for us to have the readme_example and example_custom_model.R there, but I don't know if external users will ever look there for examples etc...

Do you think the scripts there is the issue?

**Comment by nikolase90** [comment]
Might be. But I makes more sense that it is the tests. Could we just set `update = FALSE` in `testthat::expect_known_value` ?

**Comment by martinju** [comment]
> Might be. But I makes more sense that it is the tests. Could we just set `update = FALSE` in `testthat::expect_known_value` ?

Yes, but it makes it a bit more cumbersome to do package development. Also, I find it a bit strange that CRAN does not allow the default behavior of a function in this very popular package.

**Comment by nikolase90** [comment]
We'll find a workaround for the development I think.

**Comment by nikolase90** [comment]
But yeah, I agree.

**Comment by martinju** [comment]
Yes, sure. What do you suggest? 

1. add inst/scripts to .Rbuildignore, 
2. update=FASLE for expect_known_value, 
3. both 1 and 2
4. Do nothing and ask them to be more precise (but we may ask specifically whether it might be either 1 and 2)

Will surely require another revision round, but I am most keen on 4.

**Comment by nikolase90** [comment]
I would do 3. I don't think they like responses as number 4 :-)

**Comment by martinju** [comment]
> I would do 3. I don't think they like responses as number 4 :-)

Ok, I will fix that and resubmit tonight, unless you have other comments to my modifications?

**Comment by martinju** [comment]
I fixed the expect_known_value thing, but I am not able to successfully exclude the scripts folder from the built package. I tried various path versions of both the folder and individual files as found in other package's .Rbuildignore files, but every time I build the source package with RStudio, the files are still included, so I must be doing something wrong. 

Any idea @nikolase90 ?

**Comment by nikolase90** [comment]
You need to build the source first, and then install, i.e. `devtools::build()` and `R CMD INSTALL shapr_...tar.gz`. If you then look at the installed version the files will not be there.

**Comment by martinju** [comment]
> You need to build the source first, and then install, i.e. `devtools::build()` and `R CMD INSTALL shapr_...tar.gz`. If you then look at the installed version the files will not be there.

Thanks! Turned out the problem was that I was building "binary package" and not "source package", where the first for some reason does not obey .Rbuildignore...

**Comment by martinju** [comment]
Just resubmitted to CRAN with the following cran-comments:

* Revised what is specified as internal/non-internal functions, and added 
  examples such that all non-internal exported functions now have examples.
* I am not quite sure exactly what the "writing to disk" issue refers to, but
  we are using testthat::expect_known_value in our tests, which is overwriting
  certain .rds-files if and only if the test fails. I changed the default 
  behavior so that it no longer overwrites the files. We also have some
  scripts under inst/scripts that writes to disk. These are now added to 
  .Rbuildignore. Please let me know more precisely what the issue is if these
  fixes are not satisfactory.
* Deleted a call to rm(list=ls()) in a script under inst/scripts. I could not
  find any other modifications of .GlobalEnv in the vignette as suggested.
